### DISCUSSION TITLE
v1.7-exp1

### DISCUSSION THREAD

**Discussion by RDMacLachlan** [microsoft/WindowsAppSDK#4907] [Announcements]
## 1.7.0 Experimental 1 (1.7.241114004-experimental1) 🧪

WinAppSDK 1.7.241114004 Experimental 1 introduces several exciting new features designed to enhance the development experience and expand application functionality. This release includes the new CameraCaptureUI API, which simplifies the process of capturing photos and videos within your WinAppSDK app. Additionally, the `OAuth2Manager` API has been added to provide a streamlined solution for web authentication, bringing OAuth 2.0 capabilities with full feature parity across all Windows platforms. These enhancements allow developers to create more interactive, secure, and media-rich user experiences.

Furthermore, the release features the new `BackgroundTaskBuilder` API, offering integrated support for background task registration in WinAppSDK apps. This API enables developers to manage background tasks more efficiently, ensuring that applications perform essential operations even when not actively in use. Overall, WinAppSDK 1.7.0 Experimental 1 empowers developers with powerful tools to create more sophisticated, secure, and user-friendly applications, paving the way for innovative and effective software solutions.

**Check out the [release notes](https://aka.ms/windowsappsdk/1.7/1.7.0-experimental1/release-notes) for all of the new and updated features, limitations, and known issues.**

### Updating to this version

The 1.7.241114004-experimental1 package is available at: [NuGet](https://www.nuget.org/packages/Microsoft.WindowsAppSDK/1.7.241114004-experimental1)

Additionally, for those testing deployment of unpackaged apps, download the Windows App SDK Installers and MSIX packages [here](https://aka.ms/windowsappsdk/1.7/1.7.241114001-experimental1/Microsoft.WindowsAppRun7ime.Redist.1.7.zip).

## What's Changed
* Bug Fix for Cannot install version 1.3: DDLM package not found by @ssparach in https://github.com/microsoft/WindowsAppSDK/pull/4890
* [main] Update dependencies from microsoft/LiftedIXP/DCPP by @dotnet-maestro in https://github.com/microsoft/WindowsAppSDK/pull/4870
* Remove Invalid character from DevCheck Password Gen by @kythant in https://github.com/microsoft/WindowsAppSDK/pull/4903

## New Contributors
* @ssparach made their first contribution in https://github.com/microsoft/WindowsAppSDK/pull/4890

**Full Changelog**: https://github.com/microsoft/WindowsAppSDK/compare/v1.6.3...v1.7-exp1

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WindowsAppSDK/releases/tag/v1.7-exp1'>v1.7-exp1</a>.</em>

**Comment by ghost1372** [comment]
Typo issue
The 1.7.241114004-**experimental2**

**Comment by w-ahmad** [comment]
Interestingly this release is labeled as latest, not the prerelease.

**Reply by DarranRowe** [reply]
Where do you see that?

The NuGet website clearly states that it is prerelease, and Visual Studio clearly shows it is prerelease and does not show it if Include prerelease isn't checked.

**Reply by w-ahmad** [reply]
I see that in github release section.

**Reply by lostluna1** [reply]
> 我在 github 发布部分看到了这一点。

it's true

**Comment by duwenlong2** [comment]
Will the 1.7 version update the new inkcanvas?
Is there a new ink recognition library available? The recognition rate under UWP is very poor

**Reply by castorix** [reply]
What do you mean by "recognition rate is very poor" ?
I tested [Simple inking sample](https://learn.microsoft.com/en-us/samples/microsoft/windows-universal-samples/simpleink/)
and it seems to work correctly for recognition (tested in french and english)

**Reply by duwenlong2** [reply]
We tested both Chinese and English;
In a Chinese speaking environment:
The recognition rate for neat writing is 97%, and when I checked italic writing, the recognition rate for oblique writing is 90%. Writing a poem vertically was recognized as a single word;
Chinese English mixed writing: 15 English words and 75 Chinese characters. Can be recognized as a language. For example, after writing English, the remaining Chinese characters will be recognized as strange characters.
The main scene is neat writing, writing diagonally. Chinese English mixed writing.
May I ask if there have been any updates to ink handwriting on the Windows system in recent years? If so, where can I get updated logs.

**Reply by duwenlong2** [reply]
> you can file a github issue for that.

May I ask if I can submit an issue under app sdk?

**Comment by Osirisoo0O** [comment]
ApplicationTrigger still not usable with OOP task.
Can you provide a C# wasdk sample? Thanks!

**Reply by godlytalias** [reply]
Sure, Will add C# sample as well here, https://github.com/microsoft/WindowsAppSDK-Samples/pull/392

**Reply by Osirisoo0O** [reply]
And why use COM instead of windows runtime component？
Expect a reply.

**Reply by godlytalias** [reply]
For registering WinRT component for Background Task you can use Windows.ApplicationModel.Background APIs. Only for full trust COM, those APIs will throw exception (for majority of the triggers)

**Reply by godlytalias** [reply]
@Osirisoo0O I'd added C# sample here, https://github.com/microsoft/WindowsAppSDK-Samples/pull/392/commits/e123eae14bec61a1b47b1128a7322bbcc23a5ff3
Please let me know whether that helps, Thanks

**Reply by Osirisoo0O** [reply]
First，Thank you for this sample.
Perhaps there is an issue with my description.I am unable to register using ApplicationTrigger in the wasdk app.
When I use code below:
``` 
static readonly ApplicationTrigger _trigger = new();

string Name = nameof(TestTask).ToString(); 
string TaskEntryPoint =typeof(TestTask).FullName;
var builder = new Windows.ApplicationModel.Background.BackgroundTaskBuilder
{     
     TaskEntryPoint = TaskEntryPoint,
     Name = Name
};
builder.SetTrigger(_trigger);

builder.Register();
```
I got some message like that:
```
Error: System.Runtime.InteropServices.COMException (0x800700B7)
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|38_0(Int32 hr)
   at WinRT.ExceptionHelpers.ThrowExceptionForHR(Int32 hr)
   at ABI.Windows.ApplicationModel.Background.IBackgroundTaskBuilderMethods.Register(IObjectReference _obj)
   at Windows.ApplicationModel.Background.BackgroundTaskBuilder.Register()
   at App.MainWindow.RegisterTask()
```
Could you help me? thanks!

**Reply by godlytalias** [reply]
This exception would most probably because the task you're registering is not 'Windows Runtime component'. You can once check that. Also in case of Win32 apps, the same ApplicationTrigger requirements can be easily done by dispatching those tasks to threadpool threads because it don't have lifetime management issues like that of UWP. Is there some specific requirement you're trying to achieve?

**Reply by Osirisoo0O** [reply]
I am making some attempts to smooth out the differences between UWP and WASDK, in order to facilitate the migration of projects from UWP to WASDK.
There is some sample. [Osirisoo0O/WASDKSample](https://github.com/Osirisoo0O/WASDKSample)

**Comment by Gaoyifei1011** [comment]
> ## 1.7.0 Experimental 1 (1.7.241114004-experimental1) 🧪
> WinAppSDK 1.7.241114004 Experimental 1 introduces several exciting new features designed to enhance the development experience and expand application functionality. This release includes the new CameraCaptureUI API, which simplifies the process of capturing photos and videos within your WinAppSDK app. Additionally, the `OAuth2Manager` API has been added to provide a streamlined solution for web authentication, bringing OAuth 2.0 capabilities with full feature parity across all Windows platforms. These enhancements allow developers to create more interactive, secure, and media-rich user experiences.
> 
> Furthermore, the release features the new `BackgroundTaskBuilder` API, offering integrated support for background task registration in WinAppSDK apps. This API enables developers to manage background tasks more efficiently, ensuring that applications perform essential operations even when not actively in use. Overall, WinAppSDK 1.7.0 Experimental 1 empowers developers with powerful tools to create more sophisticated, secure, and user-friendly applications, paving the way for innovative and effective software solutions.
> 
> **Check out the [release notes](https://aka.ms/windowsappsdk/1.7/1.7.0-experimental1/release-notes) for all of the new and updated features, limitations, and known issues.**
> 
> ### Updating to this version
> The 1.7.241114004-experimental1 package is available at: [NuGet](https://www.nuget.org/packages/Microsoft.WindowsAppSDK/1.7.241114004-experimental1)
> 
> Additionally, for those testing deployment of unpackaged apps, download the Windows App SDK Installers and MSIX packages [here](https://aka.ms/windowsappsdk/1.7/1.7.241114001-experimental1/Microsoft.WindowsAppRun7ime.Redist.1.7.zip).
> 
> ## What's Changed
> * Bug Fix for Cannot install version 1.3: DDLM package not found by @ssparach in [Bug Fix for Cannot install version 1.3: DDLM package not found #4890](https://github.com/microsoft/WindowsAppSDK/pull/4890)
> * [main] Update dependencies from microsoft/LiftedIXP/DCPP by @dotnet-maestro in [[main] Update dependencies from microsoft/LiftedIXP/DCPP #4870](https://github.com/microsoft/WindowsAppSDK/pull/4870)
> * Remove Invalid character from DevCheck Password Gen by @kythant in [Remove Invalid character from DevCheck Password Gen #4903](https://github.com/microsoft/WindowsAppSDK/pull/4903)
> 
> ## New Contributors
> * @ssparach made their first contribution in [Bug Fix for Cannot install version 1.3: DDLM package not found #4890](https://github.com/microsoft/WindowsAppSDK/pull/4890)
> 
> **Full Changelog**: [v1.6.3...v1.7-exp1](https://github.com/microsoft/WindowsAppSDK/compare/v1.6.3...v1.7-exp1)
> 
> _This discussion was created from the release [v1.7-exp1](https://github.com/microsoft/WindowsAppSDK/releases/tag/v1.7-exp1)._

In the Windows AppSDK 1.7 roadmap there are appwindow set window size properties, but I did not find the corresponding api, when to enter Windows AppSDK 1.7?
